### PR TITLE
[FEATURE] enrich STAC metadata, add self-healing prev/next release links

### LINKED ISSUE #97: Enrich STAC catalog metadata; add self-healing prev/next release links
### Type

Task

### Scope

Multi-theme or Platform

### Skillset

engineering

## Context

Six best-practice gaps in the generated STAC catalog. Most notably, the
production build was silently dropping Table-extension fields on every
Collection — a dict reassignment in `process_theme_worker` wiped
`table:columns`, `table:row_count`, `table:primary_geometry`, and
`geoparquet:version` in non-debug mode, leaving only `features`. Other gaps:
no Collection `summaries`, no `title`/`description`/`roles` on Item assets,
Items declaring `stac_extensions: []` while using storage/alternate
extension fields, and no cross-release `prev`/`next` navigation.

`prev`/`next` are derived from the bucket's current release list at build
time — no persisted state — so releases removed by the 3-month S3 lifecycle
policy naturally drop from the chain on the next rebuild.

## Proposed change

In [src/overture_stac/overture_stac.py](src/overture_stac/overture_stac.py):

- Fix the `extra_fields` overwrite in `process_theme_worker` — merge `features` in instead of reassigning, so the four Table-extension fields survive non-debug builds.
- Populate `Collection.summaries` with per-shard min/max for `num_rows` and `num_row_groups`, computed from the Items already in memory.
- Add `title`, `description`, and `roles=["data"]` to every Item asset (top-level `aws`, `azure`, and the nested `s3` alternate).
- Add module-level `ITEM_STAC_EXTENSIONS` (`storage/v2.0.0`, `alternate-assets/v1.2.0`); stamp on every `pystac.Item` and reuse for the release-catalog's own declaration.
- Add `list_release_ids(filesystem)` helper — currently-published releases, newest-first.
- Add `link_neighbor_releases(catalog, all_release_ids, root_href)` — adds absolute `prev`/`next` URLs to a release catalog based on the current bucket contents.

In [src/overture_stac/cli.py](src/overture_stac/cli.py):

- Call `link_neighbor_releases` in both the single-release path (release-day builds) and the walk-all path (daily DAG), using `list_release_ids` to enumerate current siblings.

Regression tests in [tests/test_process_theme_worker.py](tests/test_process_theme_worker.py); new file [tests/test_link_neighbor_releases.py](tests/test_link_neighbor_releases.py) covers the self-healing contract and the single-release-day scenario.

### PR DESCRIPTION
**Pull request by jjcfrancisco**
Closes #97.

## Summary

- Fix `extra_fields` overwrite in `process_theme_worker` — production builds now keep `table:columns`, `table:row_count`, `table:primary_geometry`, `geoparquet:version`.
- Add Collection `summaries.num_rows` and `summaries.num_row_groups` rolled up from per-shard Items.
- Add `title`, `description`, `roles=["data"]` to `aws`, `azure`, and the nested `s3` alternate assets.
- Stamp `stac_extensions` (`storage/v2.0.0`, `alternate-assets/v1.2.0`) on every Item via a new module-level constant reused by the release catalog.
- New `link_neighbor_releases` helper wires `prev`/`next` between release catalogs; derived from the current bucket contents at build time — no persisted state — so releases removed by the 3-month S3 lifecycle drop from the chain on the next rebuild. Applied in both the single-release path (release-day builds) and the walk-all path (daily DAG).

## Tests

20 unit tests pass, including regression coverage for every point above and 9 tests for the neighbour-linking contract (self-heal on missing middle release, single-release-day scenario, ordering edge cases).

## Also

Adds a `justfile` for common dev commands (`just check` mirrors CI) and a two-sentence README pointer.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env just --justfile
```
[the remark this answers is not in the document]
Comment (jjcfrancisco) [reply justfile:1]: That's the entry point per documentation of `justfile`. Discarding.

File: src/overture_stac/overture_stac.py [hunk 2]
```diff
@@ -10,6 +10,56 @@
...
+ITEM_STAC_EXTENSIONS: list[str] = [
+    "https://stac-extensions.github.io/storage/v2.0.0/schema.json",
+    "https://stac-extensions.github.io/alternate-assets/v1.2.0/schema.json",
+]
+
+
+def list_release_ids(filesystem: fs.S3FileSystem) -> list[str]:
+    """Return currently-published release ids from the public bucket, newest-first."""
+    info = filesystem.get_file_info(fs.FileSelector("overturemaps-us-west-2/release"))
+    return sorted((r.path.split("/")[-1] for r in info), reverse=True)
```
[the remark this answers is not in the document]
Comment (jjcfrancisco) [reply src/overture_stac/overture_stac.py:22]: It currently cannot be used anywhere else but production, at some point this may be possible. Discarding.

File: src/overture_stac/overture_stac.py [hunk 3]
```diff
@@ -194,6 +251,7 @@ def process_theme_worker(
                                 "href": f"s3://{fragment.path}",
                                 "alternate:name": "S3",
                                 "description": "Access the files via regular Amazon AWS S3 tooling.",
+                                "roles": ["data"],
                             }
```
[the remark this answers is not in the document]
Comment (jjcfrancisco) [reply src/overture_stac/overture_stac.py:255]: I'll skip this as I think adding a title would just be "GeoParquet on AWS S3 (S3 protocol)" which seems redundant

**Review by danabauer — APPROVED** [review]

**Review by lowlydba — APPROVED** [review]
# Huge!

Amazing, can't wait to see this in action
